src/article.js

```diff
@@ -487,3 +506,30 @@ Article.prototype.handleResize_ = function() {
     this.sections[i].rerender();
   }
 };
+
+/**
+ * Removes blank paragraphs components at the start and the end of the article
+ */
+Article.prototype.trim = function() {
+  var firstComponent = this.getFirstTextComponent();
+  var lastComponent =  this.getLastTextComponent();
```

Comment:
The problem with using `getFirstTextComponent` and `getLastTextComponent` is that these methods would skip over any non-text components so that way we're no longer just trimming the beginning and the end of the article. For example this would be the result of the following:

```
<img>
<p></p>
<p>Hello</p>
<p></p>
<img>
```

Using first and last text components would skip over the images and remove blank paragraphs after and before them:
```
<img>
<p>Hello</p>
<img>
```
The result however should be the original model because there's nothing to trim from the beginning and the end of the model.

My suggestion would be to implement `getFirstComponent` (text or otherwise). And use that instead. After getting the firstComponent you can then call `component.getNextComponent` or `component.getPrevComponent` to iterate over them and stop once you hit a non-empty text component.

Does that make sens?